modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/impl/MetaStorageServiceImpl.java

```diff
@@ -82,29 +88,27 @@ public class MetaStorageServiceImpl implements MetaStorageService {
     public MetaStorageServiceImpl(RaftGroupService metaStorageRaftGrpSvc, ClusterNode localNode) {
         this.metaStorageRaftGrpSvc = metaStorageRaftGrpSvc;
         this.localNode = localNode;
+        this.executorService = Executors.newFixedThreadPool(5, NamedThreadFactory.create(localNode.name(), "metastorage-publisher", LOG));
```

Comment:
Maybe the number of CPUs?

Reply:
why, how is this pool related to the number of CPUs? It's IO bound

Reply:
Will there be any problems if the user only has 4 and 32 CPUs? Will there be starvation or underutilization of resources?

Reply:
No, this thread pool is not CPU bound, it's tied to the amount of requests to the Meta Storage. 5 is an arbitrary value, I will add a TODO to extract this value to configuration.

Reply:
done